python/cudf_polars/cudf_polars/experimental/benchmarks/pdsds_queries/q2.py

```diff
@@ -0,0 +1,228 @@
+                   sun_sales         sun_sales2,
+                   mon_sales         mon_sales2,
+                   tue_sales         tue_sales2,
+                   wed_sales         wed_sales2,
+                   thu_sales         thu_sales2,
+                   fri_sales         fri_sales2,
+                   sat_sales         sat_sales2
+            FROM   wswscs,
+                   date_dim
+            WHERE  date_dim.d_week_seq = wswscs.d_week_seq
+                   AND d_year = 1998 + 1) z
+    WHERE  d_week_seq1 = d_week_seq2 - 53
+    ORDER  BY d_week_seq1;
+    """
+
+
+def polars_impl(run_config: RunConfig) -> pl.LazyFrame:
+    """Query 2."""
+    # Load required tables
+    web_sales = get_data(run_config.dataset_path, "web_sales", run_config.suffix)
+    catalog_sales = get_data(
+        run_config.dataset_path, "catalog_sales", run_config.suffix
+    )
+    date_dim = get_data(run_config.dataset_path, "date_dim", run_config.suffix)
+    # Step 1: Create wscs CTE equivalent (union of web and catalog sales)
+    wscs = pl.concat(
+        [
+            web_sales.select(
+                [
+                    pl.col("ws_sold_date_sk").alias("sold_date_sk"),
+                    pl.col("ws_ext_sales_price").alias("sales_price"),
+                ]
+            ),
+            catalog_sales.select(
+                [
+                    pl.col("cs_sold_date_sk").alias("sold_date_sk"),
+                    pl.col("cs_ext_sales_price").alias("sales_price"),
+                ]
+            ),
+        ]
+    )
+    # Step 2: Create wswscs CTE equivalent (aggregate by week and day of week)
+    # First join with date_dim to get day names
+    wscs_with_dates = wscs.join(date_dim, left_on="sold_date_sk", right_on="d_date_sk")
+    # Create separate aggregations for each day to better control null handling
+    days = [
+        "Sunday",
+        "Monday",
+        "Tuesday",
+        "Wednesday",
+        "Thursday",
+        "Friday",
+        "Saturday",
+    ]
+    day_cols = [
+        "sun_sales",
+        "mon_sales",
+        "tue_sales",
+        "wed_sales",
+        "thu_sales",
+        "fri_sales",
+        "sat_sales",
+    ]
+    # Start with all week sequences
+    all_weeks = wscs_with_dates.select("d_week_seq").unique()
+    wswscs = all_weeks
+    for day, col_name in zip(days, day_cols, strict=False):
+        # Get sales for this specific day
+        day_sales = (
+            wscs_with_dates
+            # TODO: .filter(pl.col('d_day_name') == day) should work but does not
+            # There's some bug in cudf_polars with FILTER on Column with NULLs
+            .filter(pl.col("d_day_name").is_not_null() & (pl.col("d_day_name") == day))
+            .group_by("d_week_seq")
+            .agg([pl.col("sales_price").sum().alias(col_name)])
+        )
+        # Left join to preserve weeks with no sales for this day (which should be null)
+        wswscs = wswscs.join(day_sales, on="d_week_seq", how="left")
```

Comment:
It feels like this would be much better written as a groupby followed by a pivot.

```
wscs_with_dates.group_by("d_week_seq", "d_day_name").agg(
    pl.col("sales_price").sum().alias("sales")
).group_by("d_week_seq").agg(
    pl.col("sales").filter(pl.col("d_day_name") == day).first().alias(name)
    for day, name in zip(days, day_cols, strict=True)
)
```

Rather than this long sequence of joins. But we would need to implement filtered groupby aggregations in cudf-polars.